---
title: "Staph genome size distribuiton"
output: html_notebook
---

```{r}
library(staphopia)
library(dplyr)
library(ggplot2)
library(feather)
```


```{r}
gold_samps <- get_public_samples() %>% filter(rank == 3)
contigs <- get_assembly_stats(gold_samps$sample_id)

```

#### plasmids
 (local path will vary) These ae lists of contigs called as plasmid by SPAdes baed on coverage and other metrics
```{r}
plas_table <- read_feather("~/Downloads/staphopia-plasmids.feather") %>%
  group_by(sample_id) %>% summarise(plasmid_len = sum(length),plas_contigs = n())
```

#### create useful merged gold data table
```{r}
merged_tables <- inner_join(contigs,public, by = "sample_id") %>%
  select(sample_id,st,total_contig,total_contig_length,percent_contigs_greater_1k,l50_contig_count,n50_contig_length)
gold_summary <-inner_join(merged_tables,plas_table,by="sample_id")
```


```{r}
mean(gold_summary$total_contig_length)
sd(gold_summary$total_contig_length)
```

How does the proportion of small assemblies vary by contig length?

```{r}
p <- ggplot(gold_summary, aes(x= total_contig_length, y = percent_contigs_greater_1k)) + geom_point()
p 
```

Note the hard line at about 2.7 Mbp?
Also the group of strains with total_length greater than ~ 2.8Mbp but %contigs less than 10%
Interesting set of strains seem to all have genomes of ~ 1.8 Mbp.



```{r}
p <- ggplot(gold_summary, aes(x= total_contig_length, y = l50_contig_count, color=l50_contig_count > 50)) + geom_point()
p 
```

The l50 contig count increases with total size (above 2.75 Mb) this is probably a factor of low quality (or maybe higher levels of transposon)


```{r}
p <- ggplot(gold_summary, aes(x= total_contig_length, y = n50_contig_length)) + geom_point()
p 
```
 

Strikingly, a lot of these large genomes are ST239

```{r}
filter(gold_summary,total_contig_length < 2630000) %>%
  select(st) %>%
  table()
```


```{r}
p <- ggplot(gold_summary,aes(x=total_contig_length, color = (st==239))) + geom_histogram(binwidth=10000) 
p
```
Interesting set of peaks, especially the one over 3,000,000, which is mostly ST239


```{r}
p <- ggplot(gold_summary,aes(x=total_contig_length, color = (st==0))) + geom_histogram(binwidth=10000) 
p
```

```{r}
p <- ggplot(gold_summary, aes(x= total_contig, y = plasmid_len, color=l50_contig_count > 50)) + geom_point()
p 
```
This shows the relation between "plasmid" length, high l50 contig count (ie shredded assemblies) and genome size.  Its probable that the group of strains with large 

```{r}
p <- ggplot(gold_summary,aes(x=total_contig_length-plasmid_len)) + geom_histogram(binwidth=10000) 
p
```

```{r}
table(cut(gold_summary$total_contig_length, breaks = 100))
```
It looks like the inflexion point for the bottom limit of S. aureus genome size is 2.63 Mb.  On the other side, maybe 3.11 Mb


Start with looking at the the small ones
```{r}
under2.63Mb <- gold_summary %>% filter(total_contig_length < 2630000) 
under2.63Mb_meta <- get_metadata(under2.63Mb$sample_id)
table(under2.63Mb$st)
under2.63Mb_scccmec <- get_sccmec_type(under2.63Mb$sample_id)
summary(under2.63Mb_scccmec)
```

These are all from Sanger, all ST '0'.  However, about half are MRSA. My guess, deletions?.

Now get the large contigs in many pieces
```{r}
over3.11_small <- gold_summary %>% filter(total_contig_length > 3110000) %>%
  filter(l50_contig_count > 50)
over3.11_small_meta <- get_metadata(over3.11_small$sample_id)
table(over3.11_small$st)
```
 These strains are more diverse - no obvious common thread
 
```{r}
over3.11_large <- gold_summary %>% filter(total_contig_length > 3110000) %>%
  filter(l50_contig_count <= 49)
over3.11_large_meta <- get_metadata(over3.11_large$sample_id)
table(over3.11_large$st)

```

 Nothing stands out in these large samples
 
 ## t-tests using dplyr
```{r}
all_conlens <- gold_summary$total_contig_length
pvals <- merged_tables %>% 
  group_by(st) %>% 
  summarise(conlens = list(total_contig_length), mean_con_length = mean(total_contig_length), sd_contig_length = sd(total_contig_length)) %>% 
  filter(lengths(conlens) > 100) %>%
  rowwise() %>%
  mutate(p_value = t.test(unlist(conlens), !(all_conlens %in% unlist(conlens)))$p.value) 
  
```
 
 ## cgMLST
Check that these maybe have other S. aureus STs?
Get a list of gold St=0 and look st the cgmlst profiles

```{r}
gold_st0 <- filter(gold_summary,st == 0)
gold_st0_cgmlst <- get_cgmlst(gold_st0$sample_id)
```
How many alleles are missing?

```{r}
rowSums(gold_st0_cgmlst == 0)
rowSums(is.na(gold_st0_cgmlst))
```

This suggest no missing alleles but maybe this reflects the way null calls are counted in mentalist?

Now see if all the small genomes are similar to each other

```{r}
under_2.63Mb_cgmlt <- filter(gold_st0_cgmlst,sample_id %in% under2.63Mb$sample_id)
under2.63Mb_cgmlst_distance <- cgmlst_distance(under_2.63Mb_cgmlt)
```
```{r}
plot(hclust(as.dist(under2.63Mb_cgmlst_distance)), cex = 0.3)
```

```{r}
p <- ggplot(gold_summary,aes(x=plasmid_len)) + geom_histogram(binwidth=100000) 
p
```
Ok - there are some projects with plasmid length > 500,000.  This is definately a symptom of something amiss with the data.


```{r}
library(ggfortify)
df <- gold_summary[c(3, 4, 5, 6,7,8,9)]
autoplot(prcomp(df))
```


## SOME THOUGHTS

1.  The SPAdes plasmid desingation is an indirect measure of quality, I think.  Tends to happen more in lower quality genomes.
2. Most genomes are the "correct" size and assemble into a relatively small number of contigs.
3. There seem to be a diverse group of small genomes 

## QUESTIONS TODO
1. Detailed analysis of small and large genomes.  Are they associated with particular clades?
2. Are some clades larger/smaller than others?
3. How well do plasmid designations predict total genome size (phage?)

```{r}
session_info()
```

