---
title: "Staph genome size distribuiton"
output: html_notebook
---



```{r}
library(staphopia)
library(dplyr)
library(ggplot2)
```


```{r}
public <- get_public_samples()

```

```{r}
gold_samps <- public %>% filter(rank == 3)
contigs <- get_assembly_stats(gold_samps$sample_id)

```

```{r}
mean(contigs$total_contig_length)
sd(contigs$total_contig_length)
```

How does the proportion of small assemblies vary by contig length?

```{r}
p <- ggplot(contigs, aes(x= total_contig_length, y = percent_contigs_greater_1k)) + geom_point()
p 
```

Note the hard line at about 2.7 Mbp?
Also the group of strains with total_length greater than ~ 2.8Mbp but %contigs less than 10%
Interesting set of strains seem to all have genomes of ~ 1.8 Mbp.



```{r}
p <- ggplot(contigs, aes(x= total_contig_length, y = l50_contig_count, color=l50_contig_count > 50)) + geom_point()
p 
```

The l50 contig count increases with total size (above 2.75 Mb) this is probably a factor of low quality (or maybe higher levels of transposon)


```{r}
p <- ggplot(contigs, aes(x= total_contig_length, y = n50_contig_length)) + geom_point()
p 
```
 

Strikingly, a lot of these large genomes are ST239

```{r}
filter(merged_tables,total_contig_length < 2630000) %>%
  select(st) %>%
  table()
```

As expected?  Check that these maybe have other S. aureus STs?
Get a list of gold St=0 and look st the cgmlst profiles
```{r}
gold_st0 <- filter(gold_samps,st == 0)
gold_st0_cgmlst <- get_cgmlst(gold_st0$sample_id)
```
How many alleles are missing?

```{r}
rowSums(gold_st0_cgmlst == 0)
rowSums(is.na(gold_st0_cgmlst))
```

No missing alleles?

```{r}
p <- ggplot(merged_tables,aes(x=total_contig_length, color = (st==239))) + geom_histogram(binwidth=10000) 
p
```
Interesting set of peaks, especially the one over 3,000,000, which is mostly ST239

```{r}
table(cut(contigs$total_contig_length, breaks = 100))
```
It looks like the inflexion point for the bottom limit of S. aureus genome size is 2.63 Mb.  On the other side, maybe 3.11 Mb


```{r}
merged_tables <- inner_join(contigs,public, by = "sample_id")
```

Start with looking at the the small ones
```{r}
under2.63Mb <- merged_tables %>% filter(total_contig_length < 2630000) 
under2.63Mb_meta <- get_metadata(under2.63Mb$sample_id)
table(under2.63Mb$st)
under2.63Mb_scccmec <- get_sccmec_type(under2.63Mb$sample_id)
summary(under2.63Mb_scccmec)
```

These are all from Sanger, all ST '0'.  However, about half are MRSA. My guess, deletions?.

Now get the large contigs in many pieces
```{r}
over3.11_small <- merged_tables %>% filter(total_contig_length > 3110000) %>%
  filter(l50_contig_count > 50)
over3.11_small_meta <- get_metadata(over3.11_small$sample_id)
table(over3.11_small$st)
```
 These strains are more diverse - no obvious common thread
 
```{r}
over3.11_large <- merged_tables %>% filter(total_contig_length > 3110000) %>%
  filter(l50_contig_count <= 49)
over3.11_large_meta <- get_metadata(over3.11_large$sample_id)
table(over3.11_large$st)

```

 Nothing stands out in these large samples
 
 ## t-tests using dplyr
```{r}
merged_tables %>% group_by(st239 = st == 239) %>% 
  summarise(list(total_contig_length)) %>% 
  mutate(p.value = t.test(st239)$p.value)
```
 
 

## QUESTIONS TODO
1. Detailed analysis of small and large genomes.  Are they associated with particular clades?
2. Are some clades larger/smaller than others?
3. How well do plasmid designations predict total genome size (phage?)

```{r}
session_info()
```

